django/contrib/admin/filters.py

```diff
@@ -236,18 +236,26 @@ def __init__(self, field, request, params, model, model_admin, field_path):
         self.lookup_val = params.get(self.lookup_kwarg)
         self.lookup_val2 = params.get(self.lookup_kwarg2)
         super().__init__(field, request, params, model, model_admin, field_path)
-        if (self.used_parameters and self.lookup_kwarg in self.used_parameters and
+        if (self.used_parameters and
+                self.lookup_kwarg in self.used_parameters and
                 self.used_parameters[self.lookup_kwarg] in ('1', '0')):
             self.used_parameters[self.lookup_kwarg] = bool(int(self.used_parameters[self.lookup_kwarg]))
 
     def expected_parameters(self):
         return [self.lookup_kwarg, self.lookup_kwarg2]
 
     def choices(self, changelist):
-        for lookup, title in (
-                (None, _('All')),
+        if self.field.flatchoices:
+            list_of_choices = [(str(int(x[0])), x[1]) for x in self.field.flatchoices if x[0] is not None]
+        else:
+            list_of_choices = [
                 ('1', _('Yes')),
-                ('0', _('No'))):
+                ('0', _('No')),
+            ]
+        for lookup, title in (
+            (None, _('All')),
+            *list_of_choices
+        ):
```

Comment:
We should respect also description for `None` (test is required).

Reply:
Added the test for None as well